lib/ecto/adapters/myxql/connection.ex

```diff
@@ -1286,6 +1299,13 @@ if Code.ensure_loaded?(MyXQL) do
     defp null_expr(true), do: " NULL"
     defp null_expr(_), do: []
 
+    defp collation_expr({:ok, collation_name}, text_type)
+         when text_type in ~w/string char varchar text tinytext mediumtext longtext/a do
+      " COLLATE \"#{collation_name}\""
+    end
+
+    defp collation_expr(_, _), do: []
```

Comment:
I think we should either raise or not check the type and let the underlying SQL operation fail. Perhaps the second option is best, because I don’t know if Postgres supports collations for custom types (which would then be impossible to check).

Reply:
Postgres does support collations for custom types, so letting the SQL fail is the best option.